src/sentry/issues/issue_velocity.py

```diff
@@ -133,35 +132,38 @@ def calculate_threshold(project: Project) -> Optional[float]:
         tenant_ids={"referrer": REFERRER, "organization_id": project.organization.id},
     )
 
-    result = raw_snql_query(request, referrer=REFERRER)["data"]
-    if len(result) == 0:
-        return None
-
     try:
-        return result[0][THRESHOLD_QUANTILE["name"]]
+        result = raw_snql_query(request, referrer=REFERRER)["data"]
     except Exception:
         logger.exception(
-            "Unexpected shape for threshold query results",
-            extra={"project_id": project.id, "result": json.dumps(result)},
+            "sentry.issues.issue_velocity.calculate_threshold.error",
+            extra={"project_id": project.id},
         )
         return None
 
+    return result[0][THRESHOLD_QUANTILE["name"]]
 
-def update_threshold(project: Project, threshold_key: str, stale_date_key: str) -> float:
+
+def update_threshold(
+    project: Project,
+    threshold_key: str,
+    stale_date_key: str,
+    stale_threshold: Optional[float] = None,
+) -> float:
     """
     Runs the calculation for the threshold and saves it and the date it is last updated to Redis.
-    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is None (due to errors
-    in the query or any other reason), we save it as 0 at the shortened TTL.
+    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is None (due to
+    Snuba errors), we save it as 0 with the fallback TTL.
     """
     threshold = calculate_threshold(project)
     ttl = DEFAULT_TTL
     if threshold is None:
-        logger.error(
-            "Velocity threshold couldn't be calculated, query returned nothing",
-            extra={"project_id": project.id},
+        logger.info(
+            "sentry.issues.issue_velocity.update_threshold.fallback",
+            extra={"project_id": project.id, "fallback_value": stale_threshold},
         )
```

Comment:
could also be a counter for seeing how many times we use a stale threshold fallback and a zero fallback but i think the logs should suffice for now, open to opinions though

Reply:
Is there any reason using `logger.info` instead of `WARNING` or higher level. INFO is pretty much treated as normal operation, and won't create sentry issue.

I feel I would like to have both logging and counter.
1. For counter, it will give us metric about what's total number/percentage of queries use fallback ttl
2. The logging can give us better understanding if it impacts to all projects or limited projects (in case of project level ratelimit).

Reply:
ah the `info` instead of `warning` or a higher level was just an oversight on my part 😅  i think `warning` level is appropriate here 

for the counter, should we just count the number of queries using fallback ttl or should we count for those using default ttl as well?

Reply:
I think we should focus on operation itself, not ttl. In this case, our focus is to know we couldn't get the threshold. I think a counter let us know when the system "use the fallback threshold" v.s. "use the actual threshold" should be good enough for this